### ISSUE TITLE
something wrong with commond ` lxc config device add `

### ISSUE THREAD

**Issue by duaneya** [lxc/lxd#5267]
# Required information

 * Distribution: Ubuntu
 * Distribution version: 16.04
 * The output of "lxc info" or if that fails:
   * Kernel version: 4.15.0-38-generic
   * LXC version: 3.0.2
   * LXD version: 3.0.2
   * Storage backend in use: zfs

# Issue description
I have 4 gpu on my server and the nvidia-smi shows this:
```
Fri Nov  9 19:51:07 2018       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 410.66       Driver Version: 410.66       CUDA Version: 10.0     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  GeForce RTX 208...  Off  | 00000000:02:00.0 Off |                  N/A |
|  0%   36C    P0    65W / 260W |      0MiB / 10989MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   1  GeForce RTX 208...  Off  | 00000000:03:00.0 Off |                  N/A |
|  0%   38C    P0    58W / 260W |      0MiB / 10989MiB |      1%      Default |
+-------------------------------+----------------------+----------------------+
|   2  GeForce RTX 208...  Off  | 00000000:83:00.0 Off |                  N/A |
|  0%   35C    P0    50W / 260W |      0MiB / 10989MiB |      1%      Default |
+-------------------------------+----------------------+----------------------+
|   3  GeForce RTX 208...  Off  | 00000000:84:00.0 Off |                  N/A |
| 30%   37C    P0     1W / 260W |      0MiB / 10989MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|  No running processes found                                                 |
+-----------------------------------------------------------------------------+
```
when I use `lxc config device add test1 gpu gpu'(1),my virtual machine named  `test1`  worked well and the nvidia-smi shows just like above.

But when I try to assign only one gpu to my `test1`(use the command `lxc config device add test1 gpu0 gpu id=0`(2),there is something wrong.The `test1`'s nvidia-smi says It can not connect to driver or Unknown error.And I find something different betweed (1) and (2):

This is after (1) the `test1`:
![image](https://user-images.githubusercontent.com/12255185/48261448-12e53180-e45a-11e8-9651-ddc6cd0b7cd1.png)

and (2) don't have `nvidia0,nvidia1,nvidia2,nvidia3,nvidia-uvm,nvidia-uvm-tools` and `renderD128,renderD129,renderD130,renderD131`

I also try `lxc config device add test1 gpu0 gpu id=2`,and the output is something lisk this `can't find  /dev/nvidia`
 I think that might be problem,but I don't kown how to fix it.

**Comment by duaneya** [comment]
By the way,my card is 2080ti

**Comment by stgraber** [comment]
Any chance you can try using the LXD snap (version 3.6) instead?
We've made a number of fixes to the NVIDIA detection logic since LXD 3.0.2 was released, so that may explain what you're seeing. 3.0.3 will include those, but it's not released yet, so 3.6 may be your best bet.

You should be able to switch over using:
 - snap install lxd
 - lxd.migrate

**Comment by stgraber** [comment]
The snap also bundles libnvidia-container which may be pretty useful in your case

**Comment by duaneya** [comment]
> The snap also bundles libnvidia-container which may be pretty useful in your case

Okay,Thanks for reply.I will try that soon.

**Comment by stgraber** [comment]
Excellent, so sounds like we've got that fixed already and 3.0.3 will ship with it.

**Comment by duaneya** [comment]
> Excellent, so sounds like we've got that fixed already and 3.0.3 will ship with it.

one more thing,the   gpu id is now started from 1 to 4.When I use device add test1 gpu gpu id=0,it don’t work and also don’t report any error.if  I use device add test1 gpu gpu id=5,I will get some error information.

**Comment by stgraber** [comment]
id=0 is valid but it may refer to a non-NVIDIA GPU like some built-in graphics card or the like
